pkg/config/config.go

```diff
@@ -238,6 +277,22 @@ func (dc *DeploymentConfig) ExpandConfig() error {
 	return nil
 }
 
+// listUnusedModules provides a mapping of modules to modules that are in the
+// "use" field, but not actually used.
+func (dc *DeploymentConfig) listUnusedModules() map[string][]string {
```

Comment:
nit: personally I would prefer if this returned type `[]ModConnection` and then any processing needed happens downstream. By creating this map of lists we are introducing a new data structure that is kind of specific to the validation function.

Reply:
Not a bad idea, but it creates an import cycle if ModConnection needs to be referenced in validators (config imports validators, validators imports config).

I'm going to keep it as is for now, and consider that a follow-up improvement.